uvtools/dspec.py

```diff
@@ -1876,52 +1920,109 @@ def _fit_basis_2d(x, data, wgts, filter_centers, filter_half_widths,
+                    info['filter_params']['axis_1']['round%d'%stepnum]['filter_half_widths'] = info_t['filter_half_widths']
+                    info['filter_params']['axis_1']['round%d'%stepnum]['suppression_factors'] = info_t['suppression_factors']
+                    info['filter_params']['axis_1']['round%d'%stepnum]['basis_options'] = info_t['basis_options']
+                    info['filter_params']['axis_1']['round%d'%stepnum]['mode'] = info_t['basis'] + '_' + method
+            wgts_time = np.ones_like(wgts)
+            for i in range(data.shape[0]):
+                if info['status']['axis_1'][i] == 'skipped':
+                    wgts_time[i] = 0.
+            # restrict its time evolution in the time-direction to be consisting of time-basis functions.
+            for i, _y, _w, in zip(range(model.shape[1]), model_temp.T, wgts_time.T):
+                if np.count_nonzero(_w)/len(_w) >= skip_wgt and np.count_nonzero(_w[:max_contiguous_edge_flags]) > 0 \
+                   and np.count_nonzero(_w[-max_contiguous_edge_flags:]) >0:
+                    model_temp.T[i], _, info_t = _fit_basis_1d(x=x[0], y=_y, w=_w, filter_centers=[fc_t],
+                                                                     filter_half_widths=[fw_t],
+                                                                     suppression_factors=[sf_t],
+                                                                     basis_options=basis_options[0], method=method,
+                                                                     basis=basis, cache=cache)
+                    info['status']['axis_0'][i] = 'success'
+                else:
+                    info['status']['axis_0'][i] = 'skipped'
+                if np.any([info['status']['axis_0'][i] == 'success' for i in info['status']['axis_0']]):
+                    info['filter_params']['axis_0']['round%d'%stepnum]['method'] = info_t['method']
+                    info['filter_params']['axis_0']['round%d'%stepnum]['basis'] = info_t['basis']
+                    info['filter_params']['axis_0']['round%d'%stepnum]['filter_centers'] = info_t['filter_centers']
+                    info['filter_params']['axis_0']['round%d'%stepnum]['filter_half_widths'] = info_t['filter_half_widths']
+                    info['filter_params']['axis_0']['round%d'%stepnum]['suppression_factors'] = info_t['suppression_factors']
+                    info['filter_params']['axis_0']['round%d'%stepnum]['basis_options'] = info_t['basis_options']
+            # and subtract it (effectively subtracting rectangular Fourier model)
+            residual = (residual - model_temp) * (np.abs(wgts) > 0.).astype(float)
+            # and add the model to the overall model.
+            model += model_temp
+            # repeate this for all the rectangle bounds provided.
+    else:
+        raise ValueError("%s not a supported filt2d_mode ('rect', 'outer')"%filt2d_mode)
```

Comment:
Another place where an f-string could be used instead of the old style string formatting.